### PR TITLE
Flesh out CONTRIBUTING.md

### LINKED ISSUE #631: Add CONTRIBUTING.md
Having a CONTRIBUTING.md would help a lot IMO especially with Hacktoberfest.

Aleph is a pretty big and complicated codebase (especially if you try to understand the execution flow) so having a rough guide on

* how to set up a local dev environment
* what tests/lints to run before commit/push
* what's the general workflow once PR is made
* is it okay to touch the ci pipeline (improve/add/tweak jobs etc)
* how to handle non-app-code cases like dependency updates
* how to handle non-code cases like documentation improvements

### PR DESCRIPTION
**Pull request by KingMob**
Fixes #631

### CODE REVIEW

**Comment by KingMob** [comment]
@valerauko

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,9 +1,47 @@
...
+## Linting
+
+[clj-kondo](https://github.com/clj-kondo/clj-kondo) is recommended for linting. Because of the potemkin library and various proto-potemkin macros, Aleph can trigger a lot of false positives, and clj-kondo is the least bad at understanding them. (Handling this is ongoing; PRs welcome.) 
 
 ## Updating dependencies
 
-After you've made your changes to `project.clj`, be sure to run the tests with `lein test`!
+There’s some extra steps to test when updating dependencies:
+
+1. Obviously, after you've made your changes to `project.clj`, be sure to rerun the tests with `lein test`.
```
Comment (valerauko) [inline CONTRIBUTING.md:25]:
```suggestion
1. After you've made your changes to `project.clj`, be sure to rerun the tests with `lein test`.
```
I prefer not to use phrases like "just", "simply" or "obviously" in documentation. When I'm looking for information I couldn't figure out myself and run into a phrase like that it feels very condescending.

Comment (valerauko) [inline CONTRIBUTING.md:25]: Are developers ever expected to run the perf tests themselves?

**Review by valerauko — COMMENTED** [review]
Looks good! Just one question and one suggestion

File: CONTRIBUTING.md (hunk 1 above)
Comment (KingMob) [reply CONTRIBUTING.md]: No, those tests are just skeletons anyway. We probably, should either remove them or make them do something useful.

**Review by arnaudgeiser — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,9 +1,47 @@
 # How to contribute to Aleph
 
-This document is a work in progress!
+Thanks for helping out! Aleph is a collaborative project that wouldn’t exist without the generous time and effort of many people.
+
+This document is a work in progress, so if there's anything you feel needs to be added, pleae let us know, or file a PR.
```
Comment (DerGuteMoritz) [inline CONTRIBUTING.md:5]: Typo: "pleae"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
+3. After approving, I may ask you to clean up the branch history if it’s particularly messy.
+4. One of the reviewers will then merge.
+5. Releases should be independent of PRs; please do not change the Aleph version.
+
+## Documentation
+
+While the aleph.io literary doc site is nice, cljdoc.org has emerged as the modern standard, and we wish to support that. Please ensure that all docstrings and articles support it.
 
-Once it's ready to go, you'll need to run the `deps/lein-to-deps` script to update `deps.edn` as well. It's preferable to commit the `deps.edn` changes together with the changes to `project.clj` so the dependency changes are atomic (can be reverted in one step).
+Aleph was largely the creation of one person, and could use improvements in documentation. In particular, feel free to add comments explaining the _why_ of a decision or piece of code, even pre-existing.
```
Comment (DerGuteMoritz) [inline CONTRIBUTING.md:47]: That's a great idea :+1:

**Review by DerGuteMoritz — CHANGES_REQUESTED** [review]
Nice work :+1:

How about including a paragraph on how to deal with indentation changes? See e.g. https://github.com/clj-commons/aleph/pull/622#discussion_r954209261

File: CONTRIBUTING.md (hunk 2 above)
Comment (valerauko) [inline CONTRIBUTING.md:5]:
```suggestion
This document is a work in progress, so if there's anything you feel needs to be added, please let us know, or file a PR.
```

**Review by valerauko — APPROVED** [review]

**Review by DerGuteMoritz — APPROVED** [review]

**Pull request merged**